core/embed/rust/src/ui/component/paginated.rs

```diff
@@ -3,48 +3,86 @@ use crate::ui::{
         text::layout::{LayoutFit, TextNoOp},
         Component, ComponentExt, Event, EventCtx, FormattedText, Pad,
     },
-    display,
+    display::Color,
     geometry::Rect,
 };
 
-use super::{
-    page::{Page, PageMsg},
-    theme,
-};
+/// Implementations of `Page` wrap the component being paged. They also contain
+/// model-dependent logic like:
+///
+/// * rendering scrollbar
+/// * detecting swipe on TT
+/// * buttons for changing pages on T1
+/// * fading backlight
+pub trait Page {
+    type Content;
+    fn new(area: Rect, page: Self::Content, page_count: usize, active_page: usize) -> Self;
+    fn inner_mut(&mut self) -> &mut Self::Content;
+    fn page_count(&self) -> usize;
+    fn active_page(&self) -> usize;
+    fn fade_after_next_paint(&mut self);
```

Comment:
I think we should remove this and set the flag from the inside.

Reply:
I believe you fixed it in a64ec8a.